mantidproject/mantid
# MuonAnalysis grouping selection - change unintuitive behaviour

Fixed a bug/unintuitive behaviour reported in the MuonAnalysis interface. The choice of which group/pair to plot should not be changed unexpectedly, it should be changed only if the user explicitly chooses to (via dropdown list, or one of the Plot buttons).

This included some refactoring, a selection of free functions was turned into a class (`MuonGroupingHelper`) with the aim of making it a little clearer where the UI is updated.

**To test:**

_Test data available at_ `\\olympic\babylon5\Scratch\TomPerkins\Data\argus00060926.nxs`
1. (demonstrative test) Run through the steps in the issue and see that the behaviour is more intuitive:
   - Load the test file into MuonAnalysis. Pair `long` should be plotted by default.
   - Go to grouping tab and edit entry for `fwd` to exclude detectors 10-16, which are bad for this dataset.
   - On editing, the data should be replotted (if you have auto-update set to on in the settings tab) The pair plotted should still be `long`, not `fwd`.
   - Go back to Home tab and check the dropdown list is still on `long`.
2. Use the dropdown list on the Home tab to switch between plotting `fwd`, `bwd` and `long`. Check this works as it should.
3. On the grouping tab, use the "Plot" button by the top (groups) table. The highlighted row should be the group plotted. Check this changes the dropdown on the Home tab too.
4. As with 3, but use the "Plot" button by the bottom (pairs) table.
5. Edit groupings in the group table. The plot should update automatically (if this option is chosen in the setttings), but the previously selected group/pair (i.e. `long` by default) should not be changed.
6. As 5 with the pair table
7. Turn auto-update off in the settings and check that, under these conditions, the plot is only updated by explicitly clicking one of the "Plot" buttons
8. With `fwd` selected in the Home tab dropdown list, edit the pairs table. Check that what is plotted is still `fwd` and not `long`.
9. Try further combinations to get the interface to behave unexpectedly or break.

Fixes #15457 
- [x] The [release notes](http://www.mantidproject.org/Release_Notes_3_7_MuonAnalysis#Muon_Analysis) have been updated

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
